File: Mathematics/fibonacci/java/fibonacci.java
/*

A fibonacci series is a special series wherein every number (except for 0 and 1) is a sum of the previous 2 numbers.
example:
    0 1 1 2 3 5 8 .....
    here,
        8 = 5+3
        5 = 3+2
        3 = 2+1
A fibonacci series forms the basis of the geometric pattern that the universe forms into. Its relevance comes into play when we aggregate the series to find it matching to the golden ratio

The following program is an efficient way of finding the fibonacci series upto the Nth term via dynamic programming


*/

class fibonacci { 

	// Function to find the fibonacci series 
	static int fib(int n) 
	{ 
		int f[] = new int[n + 2];      // To store the fibonacci numbers
		int i; 
    
		f[0] = 0; 
		f[1] = 1;     // initial numbers of the fibonacci series (pre known)

		for (i = 2; i <= n; i++) { 

			// Add the previous 2 numbers 
			// in the series and store it 
			f[i] = f[i - 1] + f[i - 2]; 
		} 

		// Nth Fibonacci Number 
		return f[n]; 
	} 

	public static void
	main(String args[]) 
	{ 
		int N = 10; // taking N as 10. It can be inputted via user as well

		// Print first 10 term 
		for (int i = 0; i < N; i++) 
			System.out.print(fib(i) + " "); 
	} 
} 

File: Mathematics/fibonacci/C/fibonacci.c
/**
 * C program to find nth Fibonacci term using recursion
 */

#include <stdio.h>


/* Function declaration */
unsigned long long fibo(int num);


int main()
{
    int num;
    unsigned long long fibonacci;
    
    /* Input a number from user */
    printf("Enter any number to find nth fiboacci term: ");
    scanf("%d", &num);
    
    fibonacci = fibo(num); 
    
    printf("%d fibonacci term is %llu", num, fibonacci);
    
    return 0;
}


/**
 * Recursive function to find nth Fibonacci term
 */
unsigned long long fibo(int num) 
{
    if(num == 0)      //Base condition
        return 0;
    else if(num == 1) //Base condition
        return 1;
    else 
        return fibo(num-1) + fibo(num-2); 
}
